api/src/server/auth.rs

```diff
@@ -93,14 +144,18 @@ async fn request_token(
 }
 
 /// Provider redirects back to this route after the login
-#[get("/api/oidc/callback")]
+#[get("/api/oidc/{provider_name}/callback")]
 pub async fn route_login(
-    oidc_client: web::Data<DiscoveredClient>,
+    client_map: web::Data<ClientMap>,
+    params: web::Path<(String,)>,
     query: web::Query<LoginQuery>,
     sessions: web::Data<RwLock<Sessions>>,
     pool: web::Data<Pool>,
     identity: Identity,
 ) -> Result<HttpResponse, actix_web::Error> {
+    // TODO: handle bad name
+    let provider_name = params.0.to_string();
+    let oidc_client = client_map.map.get(&provider_name).unwrap();
```

Comment:
You don't have to do this now, but it should be pretty easy when you get to it. Just something like:
```
let oidc_client = client_map.map.get(&provider.name).ok_or(HttpRepsonse::NotFound())?
```